Freescale's latest kernel with CONFIG_PREEMPT_RT can boot up.
The cause is system ticket timer will setup irq before calling
init_hardirqs.
So the thread of ticket timer has been created. But during init_hardirqs,
it will been created again. I just modify our MSL code to setup ticket
irq with IRQF_NODELAY flag. Now our latest kernel can work with RT-patch.

Created-by: Fan YeFeng <r01011@freescale.com>

A timer interrupt should by design provide the IRQF_TIMER flag.
This flag will include IRQF_NODELAY on preempt-RT and make it start
with the correct settings on RT and non-RT.

Signed-off-by: Remy Bohmer <linux@bohmer.net>

---
 arch/arm/plat-mxc/time.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: linux-2.6.26/arch/arm/plat-mxc/time.c
===================================================================
--- linux-2.6.26.orig/arch/arm/plat-mxc/time.c	2009-03-10 22:35:50.000000000 +0100
+++ linux-2.6.26/arch/arm/plat-mxc/time.c	2009-03-10 22:38:59.000000000 +0100
@@ -178,7 +178,7 @@ static irqreturn_t mxc_timer_interrupt(i
  */
 static struct irqaction timer_irq = {
 	.name = "gpt-irq",
-	.flags = IRQF_DISABLED,
+	.flags = IRQF_DISABLED | IRQF_TIMER,
 	.handler = mxc_timer_interrupt,
 };
 
